yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -501,13 +572,124 @@ func (yb *TargetYugabyteDB) importBatch(conn *pgx.Conn, batch Batch, args *Impor
 }
 
+func (yb *TargetYugabyteDB) ImportBatchViaRecoveryMode(batch Batch, onPrimaryKeyConflictAction string,
+	args *ImportBatchArgs, exportDir string, tableSchema map[string]map[string]string) (int64, error) {
+	/*
+		Steps:
+		1. Check if batch was already imported and just failed to mark as done in last run.
+			If yes, return rows affected; caller will mark it as done.
+		2. Open the batch file
+		3. Read the batch file line by line(row) and build INSERT statment for it based on PK conflict action
+		4. [Optional] Use PreparedStatement Caching on coonPool for better performance
+		5. Execute the INSERT statement one by one
+		6. Update the Metadata about the batch imported
+		7. [Optional] log the summary about this: how many conflicts, how many inserted, how many update/upserted
+		8. Return the number of rows imported(wont necessarily be same as the number of rows in the batch)
+	*/
+
+	var rowsAffected int64
+	var err error
+	importBatchFn := func(conn *pgx.Conn) (bool, error) {
+		rowsAffected, err = yb.importBatchViaRecoverModeCore(conn, batch, args, onPrimaryKeyConflictAction)
+		if err != nil {
+			return false, err
+		}
+		log.Infof("Imported %d rows from %q", rowsAffected, batch.GetFilePath())
+		return false, nil
+	}
+	err = yb.connPool.WithConn(importBatchFn)
+	if err != nil {
+		return 0, fmt.Errorf("import batch %q: %w", batch.GetFilePath(), err)
+	}
+
+	return rowsAffected, nil
+}
+
+func (yb *TargetYugabyteDB) importBatchViaRecoverModeCore(conn *pgx.Conn, batch Batch, args *ImportBatchArgs, onPrimaryKeyConflictAction string) (int64, error) {
+	// 1. Check if the split is already imported.
+	alreadyImported, rowsAffected, err := yb.isBatchAlreadyImported(conn, batch)
+	if err != nil {
+		return 0, err
+	}
+	if alreadyImported {
+		return rowsAffected, nil
+	}
+
+	// 2. Open the batch file as datafile
+	df, err := batch.OpenDataFile()
+	if err != nil {
+		return 0, fmt.Errorf("open file %s: %w", batch.GetFilePath(), err)
+	}
+	defer df.Close()
+
+	// 3. Read the batch file line by line(row) and build INSERT statment for it based on PK conflict action
+	var rowsIgnored int64 = 0
+	rowsAffected = 0
+	insertStmt := args.GetInsertStatementBasedOn(onPrimaryKeyConflictAction)
+	for {
+		line, _, readLinErr := df.NextLine()
+		if readLinErr != nil && readLinErr != io.EOF {
+			return 0, fmt.Errorf("read line from file %s: %w", batch.GetFilePath(), err)
+		}
+		if line == "" && readLinErr != io.EOF { // TODO: rethink if this is even possible(empty line) in the batch file
+			utils.PrintAndLog("[debug] [table=%s | file=%s] empty line\n", batch.GetTableName().ForUserQuery(), filepath.Base(batch.GetFilePath()))
+			continue
+		}
+		utils.PrintAndLog("[debug] [table=%s | file=%s] line: %s\n", batch.GetTableName().ForUserQuery(), filepath.Base(batch.GetFilePath()), line)
+
+		fields := strings.Split(line, args.Delimiter)
+		stmtArgs := make([]interface{}, len(fields))
+		for i, val := range fields {
+			stmtArgs[i] = val
+		}
```

Comment:
Just realised one important thing:
Regarding the value converter, so for the CSV/Snapshot files, we do the value conversion without formatting the converted value, and for the streaming IUD events, we have a separate flag (formatIfRequired) for converting those in the respective format for Insert/Update/Delete stmts e.g.,String types are okay normal as is in CSV file but with inserts we need to have single quotes around them to make insert work.

So in recovery while using INSERT stmts for this we need the value conversion with. `formatIfRequired` flag.